### PR TITLE
Don't report error when BMC just rebooted after update

### PR DESCRIPTION
**Pull request by ofaurax**

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ linters-settings:
   revive:
     min-confidence: 0
   gocyclo:
-    min-complexity: 10
+    min-complexity: 11
```
Comment (joelrebel) [inline .golangci.yml:22]: why is this required?

File: internal/outofband/bmc.go [hunk 2]
```diff
@@ -219,6 +221,11 @@ func (b *bmc) FirmwareInstallStatus(ctx context.Context, installVersion, compone
 
 	status, err := b.client.FirmwareInstallStatus(ctx, installVersion, componentSlug, bmcTaskID)
 	if err != nil {
+		// If BMC has rebooted, task won't be found, we assume success (failure don't reboot)
+		if componentSlug == "bmc" && errors.Is(err, bmclibv2errs.ErrTaskNotFound) {
```
Comment (joelrebel) [inline internal/outofband/bmc.go:225]: use `strings.EqualFold(componentSlug, "bmc")`

**Review by joelrebel — CHANGES_REQUESTED** [review]
Thanks for the fix left a few comments

File: .golangci.yml (hunk 1 above)
Comment (ofaurax) [reply .golangci.yml:22]: Because it seems adding a new "if" in FirmwareInstallStatus makes it overflow the complexity

**Review by joelrebel — APPROVED** [review]
